flake.nix

```diff
@@ -33,108 +31,99 @@
     extra-trusted-public-keys = ["cache.garnix.io:CTFPyKSLcx5RMJKfLo5EEPUObbA78b0YQ2DTCJXqr9g="];
   };
 
-  outputs = {
+  outputs = inputs @ {
     self,
     nixpkgs,
     crane,
-    flake-utils,
-    advisory-db,
+    systems,
     rust-overlay,
-    ...
-  }:
-    flake-utils.lib.eachDefaultSystem (
-      localSystem: let
-        makePkgs = {crossSystem ? localSystem}:
-          import nixpkgs {
-            inherit localSystem crossSystem;
-            overlays = [
-              (import rust-overlay)
-              (
-                final: prev: {
-                  # TODO: Any chance this overlay will clobber something useful?
-                  rustToolchain = final.pkgsBuildHost.rust-bin.stable.latest.default.override {
-                    targets =
-                      final.lib.optionals final.stdenv.isDarwin [
-                        "x86_64-apple-darwin"
-                        "aarch64-apple-darwin"
-                      ]
-                      ++ final.lib.optionals final.stdenv.isLinux [
-                        "x86_64-unknown-linux-musl"
-                        "aarch64-unknown-linux-musl"
-                      ];
-                  };
-
-                  craneLib = (crane.mkLib final).overrideToolchain final.rustToolchain;
-
-                  inherit advisory-db;
-                }
-              )
-            ];
-          };
-
-        pkgs = makePkgs {};
+    advisory-db,
+    flake-compat,
+  }: let
+    eachSystem = nixpkgs.lib.genAttrs (import systems);
+
+    makeNixpkgs = {
```

Comment:
Usually for Nix code I reserve the term "nixpkgs" to refer to the Nixpkgs source code and use the term "pkgs" or "packages" to refer to the actual attribute set of packages instantiated from the Nixpkgs source code.  So I'd suggest calling this utility `makePkgs` instead of `makeNixpkgs` (also because it creates the `pkgs` variable, so the name would more closely match the result it is used to create).